irohad/consensus/yac/transport/impl/network_impl.cpp

```diff
@@ -67,7 +67,7 @@ namespace iroha {
         async_call_->log_->info(
             "Receive votes[size={}] from {}", state.size(), context->peer());
 
-        handler_.lock()->onState(state);
+        handler_.lock()->onState(std::move(state));
```

Comment:
Unconditional lock of the weak pointer will cause undefined behavior when a pointer type does not have a default constructor.

Please do something like
```
if (auto notifications = handler_.lock()) {
  notifications->onState(std::move(state));
} else {
  // some error handling if applicable
}
```